src/sentry/workflow_engine/handlers/detector/base.py

```diff
@@ -21,7 +21,10 @@
 class EvidenceData(Generic[T]):
     value: T
     detector_id: int
+    data_sources: list[int]
     data_condition_ids: list[int]
+    data_condition_type: Condition
+    data_condition_comparison_value: T
```

Comment:
should this be of type `T`? I think this could be `T` or a json config to access the comparator data information from the comparison data. 🤔 we may also want to have this split into 2 different fields; the value itself and then the configuration that it used on the packet to get the value.

In the end, we'll basically want a snapshot of the incoming data and the data condition at the evaluation time here.